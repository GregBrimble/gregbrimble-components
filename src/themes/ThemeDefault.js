import Theme9 from "./Theme9"

const ThemeDefault = JSON.parse(JSON.stringify(Theme9))
ThemeDefault[`name`] = `Default`

export default ThemeDefault
