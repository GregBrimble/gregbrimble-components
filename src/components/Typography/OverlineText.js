import React from 'react';

import styled from 'styled-components';
import ThemeDefault from "../../themes/ThemeDefault";

const OverlineText = styled.span`
    color: ${(props) => props.theme.neutral['900']};
    font-family: 'Open Sans', sans-serif;
    font-size: 0.75rem;
    font-weight: 600;
    line-height: 2rem;
    text-transform: uppercase;
`;

OverlineText.defaultProps = {
    theme: ThemeDefault
};

export default OverlineText;
