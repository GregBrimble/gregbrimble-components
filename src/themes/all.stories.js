import React from 'react';

import {storiesOf} from '@storybook/react';

import Theme9 from './Theme9';
import H1 from "../components/Typography/Headlines/H1";
import CaptionText from "../components/Typography/CaptionText";
import ThemeDefault from "./ThemeDefault";


function generateElement(colors) {
    return (
        <div style={{
            display: 'grid',
            gridAutoRows: '80px',
            gridGap: '10px',
            gridTemplateColumns: '15% 15% 15% 15% 15%'
        }}>
            {Object.keys(colors).sort().reverse().map(function(weight, i) {
                return (
                    <div style={{display: 'flex', height: '100%', flexDirection: 'column'}}>
                        <div style={{width: '100%', height: '100%', backgroundColor: colors[weight]}} />
                        <CaptionText>{weight} - {colors[weight]}</CaptionText>
                    </div>
                )
            })}
        </div>
    )
}

const themes = [
    Theme9,
    ThemeDefault
];

themes.map(function(theme, i) {
    storiesOf('Themes', module)
        .add(theme.name, () => {

            return (
                <div>
                    <H1>Primary</H1>
                    {generateElement(theme.primary)}
                    <H1>Neutral</H1>
                    {generateElement(theme.neutral)}
                    <H1>Supporting</H1>
                    {theme.supporting.map(function(group, i) {
                        return (
                            <div style={{marginBottom: '60px'}}>
                                {generateElement(group)}
                            </div>
                        )
                    })}
                </div>
            )
        });
});
