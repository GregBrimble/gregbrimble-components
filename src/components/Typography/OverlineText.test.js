import React from 'react';
import OverlineText from './OverlineText';
import renderer from 'react-test-renderer';
import themes from '../../themes/ThemeDefault';

describe('OverlineText', () => {
    it('renders properly', () => {
        const tree = renderer
            .create(<OverlineText>Overline Text</OverlineText>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
