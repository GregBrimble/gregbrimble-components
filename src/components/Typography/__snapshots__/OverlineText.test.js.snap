// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlineText renders properly 1`] = `
<span
  className="sc-bdVaJa cMvWkV"
>
  Overline Text
</span>
`;
